pkg/idp/provider/azure/azure.go

```diff
@@ -2,31 +2,37 @@ package azure
 
 import (
 	"context"
-	"fmt"
 	"giantswarm/dex-operator/pkg/dex"
 	"giantswarm/dex-operator/pkg/idp/provider"
 	"giantswarm/dex-operator/pkg/key"
+	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/dexidp/dex/connector/microsoft"
 	"github.com/giantswarm/microerror"
+	"github.com/go-logr/logr"
 	azauth "github.com/microsoft/kiota-authentication-azure-go"
 	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+	"github.com/microsoftgraph/msgraph-sdk-go/applications/item/removepassword"
 	"github.com/microsoftgraph/msgraph-sdk-go/models"
 	"gopkg.in/yaml.v2"
 )
 
 type Azure struct {
 	Name     string
 	Client   *msgraphsdk.GraphServiceClient
+	Log      *logr.Logger
```

Comment:
We add a logger here so that changes on the azure idp which are not reflected in the connector can still be seen in the logs